pkg/serverless/executioncontext/executioncontext.go

```diff
@@ -67,17 +67,16 @@ func (ec *ExecutionContext) SetFromInvocation(arn string, requestID string) {
 	}
 }
 
-// UpdateFromStartLog updates the execution context based on a platform.Start log message
-func (ec *ExecutionContext) UpdateFromStartLog(requestID string, time time.Time) {
+// UpdateStartTime updates the execution context based on a platform.Start log message
+func (ec *ExecutionContext) UpdateStartTime(time time.Time) {
 	ec.m.Lock()
```

Comment:
do we still need lock/unlock here

Reply:
I think since this data is written on one goroutine, and read on another, that we should lock it behind a mutex to be safe. I don't think currently there is a data race, but it would be trivial to add one without this protection.